src/overlays/actors/ovl_Dm_Tsg/z_dm_tsg.c

```diff
@@ -28,14 +28,66 @@ ActorInit Dm_Tsg_InitVars = {
     /**/ DmTsg_Draw,
 };
 
-#endif
+void DmTsg_Init(Actor* thisx, PlayState* play) {
+    DmTsg* this = THIS;
+    s32 i;
 
-extern UNK_TYPE D_06011458;
+    if (gSaveContext.save.entrance == ENTRANCE(OPENING_DUNGEON, 0)) {
+        Actor_SetScale(&this->actor, 0.1f);
+        for (i = 0; i < 100; i++) {
+            this->unk_78C[i] = (Rand_ZeroOne() * 65535.0f);
+            this->unk_856[i] = (s16)(Rand_ZeroOne() * 200.0f) + 100;
+        }
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Init.s")
+void DmTsg_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Destroy.s")
+void DmTsg_Update(Actor* thisx, PlayState* play) {
+    DmTsg* this = THIS;
+    s32 cueChannel;
+    s16 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Update.s")
+    for (i = 0; i < 100; i++) {
+        this->unk_5F8[i] = (Math_SinS((i & 0xF) * this->unk_854) * 60.0f) + 120.0f;
+        this->unk_148[i].x = Math_CosS(this->unk_78C[i]) * this->unk_5F8[i];
+        this->unk_148[i].y = this->unk_788 + (i * 16);
+        this->unk_148[i].z = Math_SinS(this->unk_78C[i]) * this->unk_5F8[i];
+        this->unk_78C[i] += this->unk_856[i];
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Draw.s")
+    this->unk_91E = false;
+    this->unk_854 += 40;
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_514)) {
+        cueChannel = Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_514);
+        if (play->csCtx.actorCues[cueChannel]->id != 1) {
+            this->unk_91E = true;
+        }
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, cueChannel);
+    }
+}
+
+void DmTsg_Draw(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    DmTsg* this = THIS;
+    s32 i;
+    u32 j;
+
+    if (this->unk_91E) {
+        for (i = 0, j = 0; i < 100; i++) {
```

Comment:
sure looks like unk_91E is a boolean marking the actor safe to draw. `canDraw` or `isDrawable`?